### DISCUSSION TITLE
How can I use `type="application/ld+json"` with my next/script Script?

### DISCUSSION THREAD

**Discussion by ajones55555** [vercel/next.js#38213] [Help] [answered]
When I run a typescript build with the following code, i get a type error: `Type error: Type '{ type: string; dangerouslySetInnerHTML: { __html: string; }; }' is not assignable to type 'IntrinsicAttributes & { children: ReactNode; }'.
  Property 'type' does not exist on type 'IntrinsicAttributes & { children: ReactNode; }'.`

```jsx
<Script
      type="application/ld+json"
      dangerouslySetInnerHTML={{
           __html: JSON.stringify(content, null, '\t')
      }}
/>
```

On the other hand, when I try to use a plain script tag in next/head, I get this error: `react_devtools_backend.js:4026 Do not add <script> tags using next/head (see inline <script>). Use next/script instead.`

How can I use the type attribute on a next/script script tag?

Thanks

**Comment by icyJoseph** [comment, marked as answer]
Hi,

Which version of next.js are you using?

```tsx
import Script from "next/script";

const content = {
  "@context": "https://json-ld.org/contexts/person.jsonld",
  "@id": "http://dbpedia.org/resource/John_Lennon",
  name: "John Lennon",
  born: "1940-10-09",
  spouse: "http://dbpedia.org/resource/Cynthia_Lennon",
};

const Home: NextPage = () => {
  return (
    <div>
      <Script
        id="app-ld-json"
        type="application/ld+json"
        dangerouslySetInnerHTML={{
          __html: JSON.stringify(content, null, "\t"),
        }}
      />
    
      Home <Link href="/blog">blog</Link>
    </div>
  );
};
```

This is working with **next@12.1.6** ~ or at least I think so

![Screenshot 2022-06-30 at 20 54 38](https://user-images.githubusercontent.com/21013447/176756120-aeff1a16-99cb-41f6-851c-1113d2c80b3b.png)

**Reply by ajones55555** [reply]
I discovered the issue was an incorrect import in our codebase - thanks for the help @icyJoseph
